terraform/projects/app-locations-api/main.tf

```
@@ -0,0 +1,176 @@
+/**
+* ## Project: app-locations-api
+*
+* Locations API node
+*/
+variable "aws_region" {
+  type        = "string"
+  description = "AWS region"
+  default     = "eu-west-1"
+}
+
+variable "stackname" {
+  type        = "string"
+  description = "Stackname"
+}
+
+variable "aws_environment" {
+  type        = "string"
+  description = "AWS Environment"
+}
+
+variable "instance_ami_filter_name" {
+  type        = "string"
+  description = "Name to use to find AMI images"
+  default     = ""
+}
+
+variable "elb_internal_certname" {
+  type        = "string"
+  description = "The ACM cert domain name to find the ARN of"
+}
+
+variable "app_service_records_alb" {
+  type        = "list"
+  description = "List of application service names that get traffic via internal alb"
+  default     = []
+}
+
+variable "asg_size" {
+  type        = "string"
+  description = "The autoscaling groups desired/max/min capacity"
+  default     = "2"
+}
+
+variable "internal_zone_name" {
+  type        = "string"
+  description = "The name of the Route53 zone that contains internal records"
+}
+
+variable "internal_domain_name" {
+  type        = "string"
+  description = "The domain name of the internal DNS records, it could be different from the zone name"
+}
+
+variable "instance_type" {
+  type        = "string"
+  description = "Instance type used for EC2 resources"
+  default     = "m5.2xlarge"
+}
+
+# Resources
+# --------------------------------------------------------------
+terraform {
+  locations_api    "s3"             {}
+  required_version = "= 0.11.14"
+}
+
+provider "aws" {
+  region  = "${var.aws_region}"
+  version = "2.46.0"
+}
+
+data "aws_route53_zone" "internal" {
+  name         = "${var.internal_zone_name}"
+  private_zone = true
+}
+
+data "aws_acm_certificate" "elb_internal_cert" {
+  domain   = "${var.elb_internal_certname}"
+  statuses = ["ISSUED"]
+}
+
+module "locations-api" {
+  source                        = "../../modules/aws/node_group"
+  name                          = "${var.stackname}-locations-api"
+  default_tags                  = "${map("Project", var.stackname, "aws_stackname", var.stackname, "aws_environment", var.aws_environment, "aws_migration", "locations_api", "aws_hostname", "locations-api-1")}"
+  instance_subnet_ids           = "${data.terraform_remote_state.infra_networking.private_subnet_ids}"
+  instance_security_group_ids   = ["${data.terraform_remote_state.infra_security_groups.sg_locations_api_id}", "${data.terraform_remote_state.infra_security_groups.sg_management_id}", "${data.terraform_remote_state.infra_security_groups.sg_aws-vpn_id}"]
```

Looking at other examples, I think this needs to use the hyphenated name. I don't think we need the `sg_aws-vpn_id` security group either, as that only seems to be used on the `backend` machine.

```suggestion
  instance_security_group_ids   = ["${data.terraform_remote_state.infra_security_groups.sg_locations-api_id}", "${data.terraform_remote_state.infra_security_groups.sg_management_id}"]
```